main.go

```diff
@@ -1,79 +1,147 @@
 package main
 
 import (
+	"context"
 	"database/sql"
+	"encoding/json"
 	"fmt"
 	"log"
+	"os"
+	"strings"
 	"time"
 
 	"gorm.io/gorm"
+
+	"github.com/sashabaranov/go-openai"
+
+	"github.com/kmesiab/chime-ai/ai/tools/transactions"
+	"github.com/kmesiab/chime-ai/database"
 )
 
 func main() {
 
+	ctx := context.Background()
+	timeoutCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+
+	var APIKEY = os.Getenv("OPENAI_API_KEY")
+	if APIKEY == "" {
+		log.Fatal("OPENAI_API_KEY environment variable not set")
+	}
+
 	var (
 		sqlDB *sql.DB
 		db    *gorm.DB
 		err   error
 	)
 
 	// Get the database connection
-	if db, err = getDBConnection(); err != nil {
+	if db, err = database.GetDBConnection(); err != nil {
 		log.Printf("Error connecting to database: %v\n", err)
 		return
 	}
 
-	// Defer the closing of the database connection
-	if sqlDB, err = db.DB(); err == nil {
+	// Clean up the database connection
+	if sqlDB, err = db.DB(); err != nil {
+		log.Printf("Error getting generic database object: %v\n", err)
+		return
+	} else {
 		defer sqlDB.Close()
 	}
 
-	// Get a repository instance
-	repository := NewTransactionRepository(db)
-
-	startDate, _ := time.Parse("2006-01-02", "2023-01-01")
-	endDate, _ := time.Parse("2006-01-02", "2024-12-31")
-
-	printTransactionsByDateRange(repository, startDate, endDate)
-	printDistinctTransactionDescriptions(repository, startDate, endDate)
-}
-
-func printDistinctTransactionDescriptions(repository *TransactionRepository, startDate, endDate time.Time) {
-	descriptions, err := repository.GetDistinctTransactionDescriptionsAndTotal(startDate, endDate)
-	if err != nil {
-		log.Printf("Error getting distinct transaction descriptions: %v\n", err)
-		return
+	repository := database.NewTransactionRepository(db)
+
+	client := openai.NewClient(APIKEY)
+
+	messages := []openai.ChatCompletionMessage{
+		{
+			Role: openai.ChatMessageRoleSystem,
+			Content: `You are a financial advisor and a SQL expert with access to a transaction
+				history database via tools and can query it for more robust data and analysis. You use database results to make 
+				informed responses to help the user.`,
+		},
+		{
+			Role:    openai.ChatMessageRoleUser,
+			Content: "Where do I spend most of my money?  Give me the top 10 places in October",
+		},
 	}
 
-	fmt.Printf("Distinct transaction descriptions between %s and %s:\n",
-		startDate.Format("2006-01-02"), endDate.Format("2006-01-02"))
-
-	for _, d := range *descriptions {
-		fmt.Printf("%s\t%2f\n", d.Description, d.TotalSpent)
+	completionRequest := openai.ChatCompletionRequest{
+		Model:    openai.GPT4oMini,
+		Messages: messages,
+		Tools:    []openai.Tool{transactions.NewTool()},
```

Comment:
Aren't you curious to see if AI will hack the db?  Let's just... yaknow... keep an eye on tihs.... :D

Reply:
I want to see if AI goes haywire and hacks the planet by giving it unfettered access to screw up our tiny unremarkable sqlite db... 

The worst it could do is gain elevated privileges to your system and propogate across the web using bitttorrent.  But I digress...